openedx/adg/lms/courseware_override/tests/test_views.py

```diff
@@ -0,0 +1,123 @@
+"""
+Tests for views.py file of courseware_override
+"""
+import mock
+import pytest
+
+from common.djangoapps.student.roles import CourseInstructorRole
+from common.djangoapps.student.tests.factories import UserFactory
+from openedx.adg.lms.applications.tests.factories import MultilingualCourseFactory, MultilingualCourseGroupFactory
+from openedx.adg.lms.courseware_override.views import (
+    get_all_course_instructors,
+    get_all_languages_for_course,
+    get_course_enrollment_count
+)
+from openedx.core.djangoapps.content.course_overviews.tests.factories import CourseOverviewFactory
+
+from .constants import DUMMY_IMAGE_URLS, END_DATE, START_DATE
+
+
+@pytest.mark.django_db
+@pytest.fixture(name='users')
+def user_fixture(request):
+    """
+    Create test users
+
+    Returns:
+        list: A list of User objects as specified
+    """
+    users = UserFactory.create_batch(request.param)
+    return users
+
+
+@pytest.fixture(scope='function', name='get_course')
+def course_fixture():
+    """
+    Creates a function to create a course with the given language
+
+    Returns:
+        func: A function that allows user to create a course with the specified language
+    """
+    def create_course(language='en'):
+        """
+        Create a course with the given language
+
+        Arguments:
+            language: The language code
+
+        Returns:
+             CourseOverview: A course overview object with the provided language
+        """
+        course = CourseOverviewFactory()
+        course.language = language
+        course.start_date = START_DATE
+        course.end_date = END_DATE
+        course.save()
+        return course
```

Comment:
This can be:
```suggestion
        return CourseOverviewFactory(language=language, start_date=START_DATE, end_date=END_DATE)
```

But if we want a generic fixture then we will do ```return CourseOverviewFactory(**kwargs)```

Reply:
Great suggestion! :100: